start_ode.sh

```diff
@@ -0,0 +1,83 @@
+#!/bin/bash
+
+# This script is used to start the Operational Data Environment (ODE) using docker compose.
+# It checks if docker is installed, sets up the .env file, spins down existing containers if they exist,
+# starts the containers, waits for the containers to start, and checks if the containers are running.
+
+
+# Check if docker is installed
+checkIfDockerInstalled() {
+  echo "Checking if docker is installed..."
+  if ! [ -x "$(command -v docker)" ]; then
+    echo "Error: docker is not installed. Install here: https://docs.docker.com/get-docker/"
+    exit 1
+  fi
+}
+
+# Set up the .env file
+setupEnv() {
+  # if .env file does not exist, create it
+  if [ ! -f .env ]; then
+    echo "Setting up .env file..."
+    if [ -z $DOCKER_HOST_IP ]
+      then
+          export DOCKER_HOST_IP=$(ifconfig | grep -A 1 'inet ' | grep -v 'inet6\|127.0.0.1' | awk '{print $2}' | grep -E '^172\.1[6-9]\.|^172\.2[0-9]\.|^172\.3[0-1]\.|^192\.168\.' | head -n 1)
+      fi
+      if [ -z $DOCKER_HOST_IP ]
+      then
+          echo "DOCKER_HOST_IP is not set and could not be determined. Exiting."
+          exit 1
+      fi
+
+      # write DOCKER_HOST_IP to .env
+      echo "DOCKER_HOST_IP=$DOCKER_HOST_IP" > .env
```

Comment:
Do we need to create the .env file first?

Reply:
We do not. Piping output to a file like this creates the file if it doesn't already exist.